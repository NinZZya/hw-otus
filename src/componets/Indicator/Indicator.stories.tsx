import React from "react";
// also exported from "@storybook/react" if you can deal with breaking changes in 6.1
import { Meta } from "@storybook/react/types-6-0";

import { Indicator, IndicatorProp } from "./Indicator";
import { Card } from "../Card/Card";

export default {
  title: "Components/Indicator",
  component: Indicator,
} as Meta;

export const EmptyIndicator: React.FC<IndicatorProp> = () => <Indicator />;
export const GreenIndicator: React.FC<IndicatorProp> = () => (
  <Indicator color="#00ff00" />
);
export const CardIndicator: React.FC<IndicatorProp> = () => (
  <Card>
    <Indicator type="card" />
  </Card>
);
