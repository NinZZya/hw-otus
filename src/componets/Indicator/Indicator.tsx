import React from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/core";

enum IndicatorType {
  CARD = "card",
}

interface DynamicStyle {
  color: string;
  margin: string;
}

export interface IndicatorProp {
  type?: "card";
  color?: string;
}

const dynamicStyle = (props: DynamicStyle) =>
  css`
    color: ${props.color};
    margin: ${props.margin};
  `;

export const IndicatorWrapper = styled.span`
  display: block;
  font-size: 2rem;
  ${dynamicStyle};
`;

export const Indicator: React.FC<IndicatorProp> = (props) => {
  const { type = "", color = "#ff0000" } = props;

  switch (type) {
    case IndicatorType.CARD:
      return (
        <IndicatorWrapper color={color} margin="0 0 0 auto">
          •
        </IndicatorWrapper>
      );
    default:
      return (
        <IndicatorWrapper color={color} margin="0">
          •
        </IndicatorWrapper>
      );
  }
};
