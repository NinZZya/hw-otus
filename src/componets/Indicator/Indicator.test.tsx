import React from "react";
import renderer from "react-test-renderer";
import { Indicator } from "./Indicator";

it("Indicator render coorectly", () => {
  const tree = renderer.create(<Indicator />).toJSON();

  expect(tree).toMatchSnapshot();
});
