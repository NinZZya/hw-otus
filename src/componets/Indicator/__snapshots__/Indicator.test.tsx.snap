// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator render coorectly 1`] = `
<span
  className="css-18r9kwf"
  color="#ff0000"
>
  â€¢
</span>
`;
